File: scripts/fetch-assets.mjs
// Fetches binary assets (fonts, ONNX model) at build time instead of committing them —
// keeps the repo/deploy payload small. Idempotent: skips any file already present.
import { existsSync, mkdirSync } from "node:fs";
import { pipeline } from "node:stream/promises";
import path from "node:path";

const ASSETS = [
  {
    url: "https://raw.githubusercontent.com/google/fonts/main/ofl/anton/Anton-Regular.ttf",
    dest: "assets/fonts/Anton-Regular.ttf",
  },
  {
    url: "https://raw.githubusercontent.com/google/fonts/main/ofl/arimo/Arimo%5Bwght%5D.ttf",
    dest: "assets/fonts/Arimo.ttf",
  },
  {
    url: "https://github.com/danielgatis/rembg/releases/download/v0.0.0/silueta.onnx",
    dest: "assets/models/silueta.onnx",
  },
];

async function fetchAsset(url, dest) {
  const fullPath = path.join(process.cwd(), dest);
  if (existsSync(fullPath)) {
    console.log(`[fetch-assets] skip (already present): ${dest}`);
    return;
  }
  mkdirSync(path.dirname(fullPath), { recursive: true });
  console.log(`[fetch-assets] downloading ${dest} ...`);
  const res = await fetch(url);
  if (!res.ok || !res.body) throw new Error(`Failed to fetch ${url}: ${res.status}`);
  const { createWriteStream } = await import("node:fs");
  await pipeline(res.body, createWriteStream(fullPath));
  console.log(`[fetch-assets] saved ${dest}`);
}

for (const asset of ASSETS) {
  await fetchAsset(asset.url, asset.dest);
}

File: automation/snob_beach_reels/fetch_assets.py
"""Downloads the OFL-licensed fonts this package needs instead of committing them — same
rationale as scripts/fetch-assets.mjs for the main Next.js app: keeps the repo/binary payload
small. Idempotent: skips any file already present. Run once after cloning:

    python3 -m automation.snob_beach_reels.fetch_assets
"""

from __future__ import annotations

import urllib.request
from pathlib import Path

from .config import FONTS_DIR

ASSETS = [
    ("https://raw.githubusercontent.com/google/fonts/main/ofl/anton/Anton-Regular.ttf", FONTS_DIR / "Anton-Regular.ttf"),
    (
        "https://raw.githubusercontent.com/google/fonts/main/ofl/archivoblack/ArchivoBlack-Regular.ttf",
        FONTS_DIR / "ArchivoBlack-Regular.ttf",
    ),
    ("https://raw.githubusercontent.com/google/fonts/main/ofl/oswald/Oswald%5Bwght%5D.ttf", FONTS_DIR / "Oswald-Bold.ttf"),
    (
        "https://raw.githubusercontent.com/google/fonts/main/ofl/caveatbrush/CaveatBrush-Regular.ttf",
        FONTS_DIR / "CaveatBrush-Regular.ttf",
    ),
    (
        "https://raw.githubusercontent.com/google/fonts/main/ofl/playfairdisplay/PlayfairDisplay%5Bwght%5D.ttf",
        FONTS_DIR / "PlayfairDisplay.ttf",
    ),
    (
        "https://raw.githubusercontent.com/google/fonts/main/ofl/playfairdisplay/PlayfairDisplay-Italic%5Bwght%5D.ttf",
        FONTS_DIR / "PlayfairDisplay-Italic.ttf",
    ),
]


def fetch_all() -> None:
    FONTS_DIR.mkdir(parents=True, exist_ok=True)
    for url, dest in ASSETS:
        if dest.exists():
            print(f"[fetch-assets] skip (already present): {dest.name}")
            continue
        print(f"[fetch-assets] downloading {dest.name} ...")
        urllib.request.urlretrieve(url, dest)
        print(f"[fetch-assets] saved {dest.name}")


if __name__ == "__main__":
    fetch_all()
